### DISCUSSION TITLE
Restore from Backup

### DISCUSSION THREAD

**Discussion by imager2019** [inventree/InvenTree#5308] [General]
I seek advice on the Backup and Restore process and best practices.

The system runs on Debian 11.7, and automatic backups are occurring through Inventree daily (currently, a new backup occurs every 25th hour - not sure why not 24 hours, but each day, the backup occurs precisely 25 hours after the last backup).

The backups consist of two files located in the `/opt/inventree/data` folder and are named:
```
debian-2023-07-20-180029.tar.gz
default-debian-2023-07-20-180028.dump.gz
```

I want to clone the current site to a VM Workstation (running same OS version) to play with some features away from the production site. I am unable to perform a restore on the VM site.

When I execute the `invoke restore` (from my `/opt/inventree` path), I receive the following, but I cannot interpret where I may be going wrong.
```
(env) inventree@debian:/opt/inventree$ invoke backup
Backing up InvenTree database...
load version from file
Traceback (most recent call last):
  File "/opt/inventree/InvenTree/manage.py", line 23, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/inventree/env/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/inventree/env/lib/python3.9/site-packages/django/core/management/__init__.py", line 363, in execute
    settings.INSTALLED_APPS
  File "/opt/inventree/env/lib/python3.9/site-packages/django/conf/__init__.py", line 82, in __getattr__
    self._setup(name)
  File "/opt/inventree/env/lib/python3.9/site-packages/django/conf/__init__.py", line 69, in _setup
    self._wrapped = Settings(settings_module)
  File "/opt/inventree/env/lib/python3.9/site-packages/django/conf/__init__.py", line 170, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/inventree/InvenTree/InvenTree/settings.py", line 116, in <module>
    STATIC_ROOT = config.get_static_dir()
  File "/opt/inventree/InvenTree/InvenTree/config.py", line 230, in get_static_dir
    raise FileNotFoundError('INVENTREE_STATIC_ROOT not specified')
FileNotFoundError: INVENTREE_STATIC_ROOT not specified
```

From my eye, it looks like the **INVENTREE_STATIC_ROOT** is missing - but I have no idea what this should be, where to edit, etc.
Would appreciate anyone with experience in restoring for advice and thoughts. I am stuck where to look next.

John

**Comment by SchrodingersGat** [comment]
John, check out the docs here - https://docs.inventree.org/en/latest/start/config/?h=inventree_static_root#file-storage-locations - should be what you are looking for

**Reply by imager2019** [reply]
Thanks for pointing me at that note.
Could you help me where to set these?

I have a pair of `config.yaml` files on my system (Debian 11). One in `etc/inventree` and the other in `opt/inventree/InvenTree`, so not sure which one to edit (the one in `opt/inventree/InvenTree` is locked, but I can edit as Admin.
![image](https://github.com/inventree/InvenTree/assets/49574724/d336b0df-9a48-4ccf-a517-0ff2c2edc26d)

The backup_dir: is currently commented out:
```
### Backup configuration options ###
# INVENTREE_BACKUP_DIR is the local filesystem location for storing backups
backup_storage: django.core.files.storage.FileSystemStorage
#backup_dir: '/home/inventree/data/backup'
#backup_options:
```
Is it as simple as removing the `#` and ensuring the path exists? If so, can I try that and reboot the server to reload the new settings?

Is there also a command that I can issue to see how all the Environment variables are set to ensure they are in place?


John

**Reply by imager2019** [reply]
I think I am making progress. I performed the edits and updated the database type - seems to be working. Will reach out of I still have issues...

**Reply by imager2019** [reply]
I did some experimenting this evening. I can get the backup to run, but the backups appear to be empty files. I am, however, getting daily backups generated from the system automatically, and these are put into /opt/inventree/data/backup and run once per day. When they run I get a `debian-2023-07-22-210019.tar.gz` and a `default-debian-2023-07-22-210019.dump.gz` where the .tar.gz file contains the label, part_files, part_images and report folders with the files from the system.

In digging a bit deeper, my database file is located in `opt/inventree/data` folder and is approximately 16.6MB. In the `config.yaml` file
it is looking for the database at:

  ENGINE: sqlite3
  NAME: '/home/inventree/database.sqlite3'

I have attempted to adjust the NAME to `NAME: '/opt/inventree/database.sqlite3'`, but still get the empty backup.

Am I setting the path correctly?

I am also now questioning whether the MEDIA_ROOT, STATIC_ROOT, and BACKUP configurations are set correctly as follows:

```
media_root: '/home/inventree/data/media'

static_root: '/home/inventree/data/static'

backup_storage: django.core.files.storage.FileSystemStorage
backup_dir: '/home/inventree/data/backup'
```

Have I really messed things up by not setting paths initially when I set things up? 

Not sure where to look next... Thoughts??

**Reply by SchrodingersGat** [reply]
@imager2019 the media and static directories need to point to locations which actually exist on your system. If you are following the installer, it should do that for you, but if you are doing a custom / manual setup, you do need to take care that the dirs are in the right places
